Mage.Sets/src/mage/cards/s/SkullRend.java

```diff
@@ -38,44 +34,31 @@ public SkullRend copy() {
 
     private static class SkullRendEffect extends OneShotEffect {
 
-        public SkullRendEffect() {
+        private SkullRendEffect() {
             super(Outcome.Damage);
             staticText = "{this} deals 2 damage to each opponent. Those players each discard two cards at random";
         }
 
-        public SkullRendEffect(final SkullRendEffect effect) {
+        private SkullRendEffect(final SkullRendEffect effect) {
             super(effect);
         }
 
         @Override
         public boolean apply(Game game, Ability source) {
-            Player controller = game.getPlayer(source.getControllerId());
-            if (controller != null) {
-                for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
-                    if (!Objects.equals(playerId, source.getControllerId())) {
-                        Player opponent = game.getPlayer(playerId);
-                        if (opponent != null) {
-                            // damage
-                            opponent.damage(2, source.getSourceId(), game);
-                            // discard 2 cards at random
-                            int amount = Math.min(2, opponent.getHand().size());
-                            for (int i = 0; i < amount; i++) {
-                                Card card = opponent.getHand().getRandom(game);
-                                if (card != null) {
-                                    opponent.discard(card, source, game);
-                                }
-                            }
-                        }
-                    }
+            for (UUID playerId : game.getOpponents(source.getControllerId())) {
+                Player opponent = game.getPlayer(playerId);
+                if (opponent == null) {
+                    continue;
                 }
-                return true;
+                opponent.damage(2, source.getSourceId(), game);
+                opponent.discard(2, true, source, game);
```

Comment:
Maybe card's bug: I think there should be two cycles (damage all first, then discard all -- as card's rules).